### ISSUE TITLE
Support Ubuntu 16.04

### ISSUE THREAD

**Issue by obrienmd** [flynn/flynn#2789]
Currently, upon running manual install of Flynn on Ubuntu 16.04, the installer fails noting that 14.04 is required. Given 16.04 is an LTS release, is Flynn planning on supporting it in the near term?

**Comment by lmars** [comment]
@obrienmd the restriction to Ubuntu 14.04 is simply because that is the version we have been using to develop and test Flynn, so we are confident that Flynn runs well on that version (Flynn doesn't really have explicit dependencies on that version).

We may upgrade our development and testing environments to 16.04, but it isn't our highest priority right now.

For actually running on 16.04, you are more than welcome to try, just comment [these lines](https://github.com/flynn/flynn/blob/master/script/install-flynn.tmpl#L30-L32) in the `install-flynn` script before running it.

**Comment by obrienmd** [comment]
Will the lines adding ZFS repos cause issues as well, given ZFS is in repos shipped with 16.04?

**Comment by lmars** [comment]
@obrienmd perhaps, I think it's a case of running the script and seeing what happens.

**Comment by obrienmd** [comment]
I commented out the 14.04 test lines, the ZFS repo / linux headers add lines, and replaced ubuntu-zfs (zfs module ships with 16.04) with zfsutils-linux in the packages var. Installs fine, but the init script doesn't work. I was hoping 16.04's compatibility layer for upstart would do this automagically, but apparently not. I may dig more on this soon, or I may just revert to 14.04 for the new Flynn cluster :)

**Comment by obrienmd** [comment]
@lmars I reverted to 14.04.4 - thanks for the feedback - feel free to close this of course, unless you want it open to track 16.04 move :)

**Comment by lmars** [comment]
@obrienmd ok thanks for giving it a try, we will leave this open for reference, we will likely make the switch at some point.

**Comment by obrienmd** [comment]
Thanks!

**Comment by jojobo** [comment]
@lmars . It would be great if you support ubuntu 16.04 because installing ZFS is giving issues on some vps as scaleway, see https://github.com/scaleway/kernel-tools/issues/257. So to install flynn on such vps ubuntu 16.04 support is needed. Thanks.

**Comment by lmars** [comment]
Due to #2459, updating to 16.04 for our development environment will require us to either install Docker 1.9.1 manually (1.9.1 is not available in Docker's xenial apt repository), or support the latest version of Docker.

**Comment by wobblybobz** [comment]
i went through and created a flynn-host.service systemd file and modified the install-flynn script to support 16.04.

run 



`flynn-host init` and it completes fine after downloading all the stuff for Layer 0
then i run systemctl start flynn-host the service starts.
when i run `CLUSTER_DOMAIN=example.com flynn-host bootstrap`
the script starts as below...
```
check online-hosts
require-env require-env
resouce-check resource-check
run-app discoverd
run-app flannel
wait-hosts wait-hosts
gen-random pg-password
gen-random pg-password _password is here_
run-app postgres
run-app postgres error: bootstrap: host error while launching job: "Unable to set up networking: network is unreachable"
```

[flynn-16.04.zip](https://github.com/flynn/flynn/files/351335/flynn-16.04.zip)

**Comment by lmars** [comment]
@wobblybobz thanks for giving it a try!

The bootstrap error is because Ubuntu 16.04 ships with a newer version of libvirt and there are some incompatibilities, but we are currently working to move away from libvirt (#3030) so it will be worth trying this again once that change has been made.

**Comment by alessiodionisi** [comment]
I also am trying to use Flynn on scaleway and I believe that compatibility with 16.04 is the only way to do it

**Comment by thomasfedb** [comment]
Being able to confidently deploy on 16.04 (or CentOS) would be very helpful for an upcoming deployment I'm currently planning. University datacentre policies must be dealt with.

**Comment by wobblybobz** [comment]
woot woot.... 
Just tried modifying the installer script and installing Flynn again on 16.04 and it works...
here are the files i used... https://gist.github.com/wobblybobz/5a22835222621f073370f40024399879
Step 1: Copy the flynn-host.service file to /etc/flynn/flynn-host.service 
Step 2: Run the install-flynn script
Step 3. Follow the rest of the instructions from the manual install page.

The flynn-host bootstrap worked and the service has started and the dashboard is accessible.
I have not loaded another app just yet but will be doing this shortly

**Comment by wobblybobz** [comment]
Further update, I have been able to build the sample apps for node, python, ruby and PHP on Flynn and After building the vagrant development system box and editing the docker.go file for the CLI to remove the --email tag from the login function and built the Flynn CLI. I Installed Docker from the get.docker.io/ubuntu repo currently have version 1.12.1 installed did the docker build and then flynn docker push as per the Docker docs example and it started and seems to be working also.

**Comment by lmars** [comment]
@wobblybobz thanks for your work on this!

We are aware of the Docker issues (#3466) and are waiting until we migrate to SquashFS based images (#3100) before we can drop the dependency on Docker <= 1.9.1 (#2459), which we will be doing in the next few weeks.

**Comment by tuukkamustonen** [comment]
Sorry for hijacking the ticket, but this is related: what about debian 8, is it supported or are there plans to support it?

**Comment by lmars** [comment]
@tuukkamustonen we have no immediate plans to update the install script for Debian, although there is no reason Flynn won't actually run on Debian so feel free to try modifying it yourself:

https://dl.flynn.io/install-flynn

The main change required is installing ZFS + AUFS (we will be dropping AUFS soon though):

https://github.com/flynn/flynn/blob/master/script/install-flynn.tmpl#L120-L144

**Comment by ianwalter** [comment]
How about adding a `requirements` section to the docs specifying which OS(s) are supported?

**Comment by bumpmann** [comment]
If you update it, can you also update the image on the aws / etc. install ?

**Comment by bdevore17** [comment]
Any idea when 16.04 will be officially supported?

**Comment by titanous** [comment]
The manual install script now supports 16.04, this will be available after the next nightly release. We will update the AMIs and images used by `flynn install` after the upcoming stable release next week.

**Comment by wasi** [comment]
@titanous thanks alot! I will test it as soon as its available.

**Comment by bumpmann** [comment]
great :)

**Comment by regedarek** [comment]
it is still open?

**Comment by titanous** [comment]
@regedarek This issue will be closed once we move our VM images to 16.04
